chainer/function_node.py

```diff
@@ -589,3 +591,232 @@ def delete_hook(self, name):
 
         """
         del self.local_function_hooks[name]
+
+
+def grad(outputs, inputs, grad_outputs=None, grad_inputs=None, set_grad=False,
+         retain_grad=False, enable_double_backprop=False):
+    """Computes the gradient of output variables w.r.t. the input variables.
+
+    This function implements the backpropagation algorithm. While
+    :meth:`Variable.backward` also implements backprop, this function selects
+    the smallest paths in the computational graph needed to compute the
+    gradients w.r.t. inputs. The error is backpropagated only through these
+    selected paths, which may reduce the overall computational cost.
+
+    This function also differs from :meth:`Variable.backward` in the way to
+    return the gradients; it directly returns the gradient variables as a list
+    instead of setting gradients to the :attr:`Variable.grad_var` attribute of
+    the original variable. It means users do not need to clear the gradient
+    w.r.t. each variable before computing the gradient using this function.
+    If ``set_grad`` option is set to ``True``, the computed gradient is also
+    stored in the :attr:`Variable.grad_var` attribute of each variable, whereas
```

Comment:
Consider rewording:

If ``set_grad`` option is set to ``True``, the computed gradient is also
    stored in the :attr:`Variable.grad_var` attribute of each variable, in which case
    any original value of :attr:`Variable.grad_var` will be updated even if it had already
    been set.